cli: Add a `create kubeconfig` command to help working with control planes

ironcladlou opened the pull request:
This commit introduces a `create kubeconfig` command to the CLI which renders a
kubeconfig with a context for every HostedCluster resource. The kubeconfig for
each cluster is based on the secret referenced by the status of the
HostedCluster.

This is a helper meant to be incorporated into other CLI workflows.

csrwng commented:
@ironcladlou this is awesome. Just a couple of suggestions... 

1. If there are no kubeconfigs, it would be great to get a warning in stderr that no kubeconfigs were found. Otherwise, you think you have a kubeconfig but it's really empty.
2. If there's only one kubeconfig (which in our case is likely the 80% case), set its context as the current context. Otherwise you get something like below when trying to use the kubeconfig:
    ```
    error: Missing or incomplete configuration info.  Please point to an existing, complete config file:


    1. Via the command-line flag --kubeconfig
    2. Via the KUBECONFIG environment variable
    3. In your home directory as ~/.kube/config

    To view or setup config directly use the 'config' command.
    ```
    Until you do `oc config use-context [blah]`

ironcladlou commented:
@csrwng in my own usage patterns I don't rely on a default context but it seems harmless to add, so I incorporated setting a default context and added some logging. I went ahead and pushed the changes but still need to test for regressions.

ironcladlou commented:
Tested, no regressions

csrwng commented:
/lgtm
